implement-directus-auth-in-nextjs-14/index.md

```diff
@@ -0,0 +1,995 @@
+---
+
+## Introduction
+In this tutorial, you will learn how to implement Authentication on a Next.js blog website using Directus as a Headless CMS. You’ll register new users, login, handle refresh tokens, create posts, list all posts, and edit/delete posts where the auth user is the post creator.
+
+## Before You Start
+To follow along with this tutorial, you will need the following knowledge and tools:
+
+- [Node.js](https://nodejs.org/en/download/) v18 and above installed on your computer.
+- A Directus project - follow our [quickstart guide](https://docs.directus.io/blog/docs.directus.io/getting-started/quickstart.html) if you don't already have one.
+- Prior experience with Next.js.
+
+<!-- ## Your Sections Here -->
+## Allowing Public Role To Create Users
+
+1. Navigate to `Settings`.
+2. Select `Access Control`.
+3. Click on the `Public Role`.
+4. Access the `System Collection`.
+5. Locate `directus_users`.
+6. Under the `Create` option, choose `Use Custom`.
+7. Enable field permissions for `first_name`, `last_name`, `role`, `Email` and `Password` options.
+![Directus public role field permission settings](public-role-field-permission-settings.png)
+8. Enable field validation where `role` equals the Author role `Primary Key`
+![Directus public role field validation settings](public-role-field-validation-settings.png)
+
+## Creating posts collection
+
+1. Navigate to `Settings`.
+2. Select `Data Model`.
+3. Create a new collection named `posts`.
+4. Check the fields that you want returned on the `posts` data
+![Directus posts optional fields settings](Directus-posts-optional-fields-settings.png)
+5. Create additional fields for the post:
+
+- title => Input field
+- content => WYSIWYG
+- image => Image
+The `title` and `content` fields should be required.
+![Directus post additional fields settings](Directus-post-additional-fields-settings-1.png)
+
+## Creating a New User Role
+
+1. On the Access Control page, click the plus button.
+2. Name your new role, as `Author`.
+3. Under the "Posts" collection, enable `Create` and `Read`.
+4. For `Edit` and `Delete`, select `Use Custom`.
+5. Add Filter `user_created -> id Equals $CURRENT_USER.id` for both update and delete options.
+6. This configuration ensures that users can read all posts but are restricted to updating and deleting only their own posts.
+7. Retrieve the `Primary Key` associated with the `Author` role by selecting the `Author Role` option and then tapping the information icon where you'll be able to copy the `Primary Key`
+8. On the Access Control page, click `Public`. Click `create` for `directus_users` from the options, then click on `Use custom`. Proceed to `field presets` and paste the Primary Key.
+
+By following these steps users created by the public role will be given the `Author` role.
+
+Add the following to your `.env` file:
+
+```text
+USER_ROLE=USER_ROLE_ID_FROM_YOUR_DIRECTUS_PROJECT
+```
+
+## Setting up Layout Configuration
+
```

Comment:
Can this be removed / minimized and just straight into registration? It feels like a lot of boilerplate.

Perhaps start with registration, then login, then altering the UI based on state (the visible nav links), and then protecting routes. Read, create, update, and, finally, delete.

Reply:
@phazonoverload This is an essential piece of the Next.js App. Removing it will result in an error after a successful login. I'll minimize it instead.

Reply:
Minimizing is fine :)